stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/TextDecorator.kt

```diff
@@ -0,0 +1,80 @@
+package io.getstream.chat.android.ui.messages.adapter.viewholder.decorator
+
+import android.widget.TextView
+import androidx.core.content.ContextCompat
+import com.getstream.sdk.chat.adapter.MessageListItem
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.messages.adapter.viewholder.GiphyViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.MessagePlainTextViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyFileAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.OnlyMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithFileAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.adapter.viewholder.PlainTextWithMediaAttachmentsViewHolder
+import io.getstream.chat.android.ui.messages.view.MessageListItemStyle
+
+internal class TextDecorator(val style: MessageListItemStyle) : BaseDecorator() {
+
+    private fun decorateMessageText(
+        textView: TextView,
+        isMine: Boolean,
+    ) {
+        val defaultTextColor = ContextCompat.getColor(textView.context, R.color.stream_ui_black)
```

Comment:
If you keep those attributes in VHs, you could just provide the required color as an argument to this method

Reply:
I supposed that's true - just doesn't seem to be the domain of the VH to do that since we delegate all of the styling to decorators. The background decorator, for example, contains default colors in a companion.